misc/pytorch_toolkit/miriad_compression_codes/src/utils/dataprep.py

```diff
@@ -0,0 +1,109 @@
+import os, re, numpy as np
+from medpy import io
+from PIL import Image
+from skimage.util import view_as_windows
+
+# Defining criteria to extract patches
+PATCH_UP_BOUND = 0.7
+PATCH_LOW_BOUND = 0.3
+
+def consider_patch( patch ):
+	counts, _ = np.histogram(patch * 255., bins=256)
+	if counts[0] < sum(counts)/3 and \
+		counts[-1] < sum(counts)/3 \
+		and	patch.mean() > PATCH_LOW_BOUND and patch.mean() < PATCH_UP_BOUND:
+		return True             
+	else:
+		return False
+
+def main( args ):
+	# counters for gathering statistics
+	train_count, test_count = 0, 0
+	unint16_count, uint8_count = 0, 0
+	all_patch_count, valid_patch_count = 0, 0
+
+	try:
+		for root, folders, files in os.walk(os.path.abspath(args.path_to_dataset)):
+
+			# only folders containing just one .dcm (000000.dcm) are of interest
+			if len(files) == 1:
+				dcm_path = os.path.join(root, files[0])
+				print(dcm_path)
+				# get the patient ID and training indicator
+				patient_id = re.findall(r"\D(P_\d{5})\D", dcm_path)[0] # search for 'P_?????' pattern
+				is_training = 'Training' in dcm_path
+
+				image, _ = io.load(dcm_path) # medpy's .dcm loader
+				image = np.squeeze(image) # squeeze out '1' length dimensions
+
+				# only 16bit mammograms are considered
+				if image.dtype == 'uint16' and \
+					image.shape[0] > args.patch_dim and image.shape[1] > args.patch_dim:
+					# image should be atleast as big as the window (patch_dim x patch_dim)
+
+					unint16_count += 1
+					image = image.astype(np.float32) / (2**16 - 1) # normalize b/w [0,1]
+
+					if is_training:
+						# Extract (non-overlapping) patches for training
+						patches = view_as_windows(image, window_shape=(args.patch_dim, args.patch_dim), step=args.patch_dim)
+						for rowid, row in enumerate(patches):
+							for colid, patch in enumerate(row):
+								all_patch_count += 1
```

Comment:
Indentation looks a bit weird. Check if it is proper.

Reply:
corrected the indentation